### ISSUE TITLE
.keepwithnext and figures fallback interaction

### ISSUE THREAD

**Issue by andreacampi** [Treesaver/treesaver#290]
tl;dr I think that Page.fillColumn should skip fallback for figure that have already been placed when looking for the next block.

Consider an article like this:

```
<article>
  <h1 class="keepwithnext">Section Title</h1>

  <figure>
    <div data-sizes="double"><img src="image1"></div>
    <div data-sizes="fallback">
      <p>Short text</p>
    </div>
  </figure>

  <figure>
    <div data-sizes="double">….</div>
    <div data-sizes="fallback">
      <p>Text 1</p>
      <p>Text 2</p>
  </div>
  </figure>
</article>
```

And a very simple grid with a container for the double plus two columns.

As long as the height is sufficient to fit everything, this works nicely; this example is too simple to show it off, but .keepwithnext really works nicely with figure fallback content, too.

As you make the browser windows shorter, this breaks down and you end up with the "Section Title" in one column and "Text 1/Text 2" in the second one.
This happens because of an inconsistency in Page.fillColumn:
- block 0 (Section Title) is not processed yet, its nextSibling is the next figure; when we reach [Page#L509](https://github.com/Treesaver/treesaver/blob/master/src/layout/page.js#L509) we'll check for available space using its firstLine, and it may very well fit. If that happens we continue the loop, adding the text to the column and `continue`ing the loop;
- block 1(<p>Short text</p>) is a figure fallback and it's already processed, so the check at line 442 will pass and we'll skip over to the next block;
- block 2 is much taller than we expect and it won't fit! and by now it's too late to undo block 0.

**Comment by andreacampi** [comment]
See https://github.com/andreacampi/treesaver/tree/issue-290 for my attempt at fixing this.
Existing tests pass; writing a specific test for this sounds like a daunting task though.

I'm not sure this is a complete fix. Page.nextNotUsedBlock probably needs to use getNextNonChildBlock instead of simply iterating on blocks?

**Comment by andreacampi** [comment]
Fix merged.
